### PR TITLE
Persist server settings in DB, added separate class for per-guild preferences [#78]

### PR DESCRIPTION
**Pull request by taahamahdi**
Deferred GameSession creation to game start [#85]

### CODE REVIEW

File: kmq.js [hunk 1]
```diff
@@ -2,37 +2,53 @@ const Discord = require("discord.js");
...
     console.log(`Logged in as ${client.user.tag}!`);
 });
 
 client.on("message", (message) => {
     if (message.author.equals(client.user) || message.author.bot) return;
-    if (!gameSessions[message.guild.id]) {
-        gameSessions[message.guild.id] = new GameSession();
+    if (!guildPreferences[message.guild.id]) {
+        guildPreferences[message.guild.id] = new GuildPreference();
+        let guildPreferencesInsert = `INSERT INTO guildPreferences VALUES(?, ?)`;
```
Comment (Brainicism) [inline kmq.js:23]: Move this into a different function

File: kmq.js [hunk 2]
```diff
@@ -2,37 +2,53 @@ const Discord = require("discord.js");
...
+            command.call({
+                client,
+                gameSessions,
+                guildPreference,
+                message,
+                db,
+                parsedMessage,
+                botPrefix
+            });
+            let guildPreferencesUpdate = `UPDATE guildPreferences SET guildPreference = ? WHERE guildID = ?;`;
```
Comment (Brainicism) [inline kmq.js:44]: Do the guild preference updates in the commands that update the preferences

File: kmq.js [hunk 3]
```diff
@@ -78,6 +94,22 @@ const parseMessage = (message, botPrefix) => {
         console.error("No bot token set. Please update config.json!")
         process.exit(1);
     }
+    let guildPreferencesTableCreation = `CREATE TABLE IF NOT EXISTS guildPreferences(
+        guildID TEXT NOT NULL,
+        guildPreference JSON NOT NULL
+    );`;
+
+    db.query(guildPreferencesTableCreation).catch((err) => {
```
Comment (Brainicism) [inline kmq.js:102]: Should `await` these to ensure they get executed in order

File: kmq.js [hunk 4]
```diff
@@ -78,6 +94,22 @@ const parseMessage = (message, botPrefix) => {
         console.error("No bot token set. Please update config.json!")
         process.exit(1);
     }
+    let guildPreferencesTableCreation = `CREATE TABLE IF NOT EXISTS guildPreferences(
```
Comment (Brainicism) [inline kmq.js:97]: Convention is underscore-case instead of camelcase

Comment (sabihismail) [reply kmq.js:97]: this is factual

**Comment by Brainicism** [comment]
@sabihismail Does this look good to you?

**Pull request merged**